phenol-core/src/main/java/org/monarchinitiative/phenol/ontology/data/TermSynonym.java

```diff
@@ -32,11 +45,61 @@
    * @param synonymTypeName Optional synonym type name, <code>null</code> if missing.
    * @param termXrefs Optional dbxref list, <code>null</code> if missing.
    */
-  public TermSynonym(String value, TermSynonymScope scope, String synonymTypeName, List<TermXref> termXrefs) {
+  public TermSynonym(String value, TermSynonymScope scope, String synonymTypeName, List<TermXref> termXrefs, String synType) {
     this.value = value;
     this.scope = scope;
     this.synonymTypeName = synonymTypeName;
     this.termXrefs = termXrefs;
+    if (synType != null && ! synType.isEmpty()) {
```

Comment:
Wouldn't this be better placed in the SynonymType class? Then TermSynonym should take a SynonymType in the constructor.

Reply:
One way of implementing this would be to put the enum into its own file and then to have a factory method that would have as its default a SynonymType.NONE. Is this what you are suggesting? I think it would be fine, but it is another implementation detail that we expose, currently, just the isLayperson() methods are exposed.
This would be so much easier with Java 9/11 modules haha!